### ISSUE TITLE
Dropbox catalog module

### ISSUE THREAD

**Issue by Phyks** [ampache/ampache#1300]
# Environment
- Ampache version: develop
- Web server + version:
- Server operating system:
- Client operating system:

- Client type:
  - Web
# Settings

# Description

- [ ] The doc on catalog creation is not up to date with current way to obtain a Dropbox API key.
- [ ] There is no progressbar in the web player when playing songs from a Dropbox backend.
# Logs

**Comment by Phyks** [comment]
Note: Dropbox is closing API v1. I am not sure, but most likely the dropbox module will stop working soon, isn't it?

**Comment by leonardo-chronicles** [comment]
Yes, as of 6/28/2017 v1 will be turned off [Dropbox announced](https://blogs.dropbox.com/developers/2016/06/api-v1-deprecated/).

V2 is the one currently online and working.

I guess an alternative to using the Catalog Type **Dropbox**, instead, is to use the Catalog Type **Local**.

Dropbox provides for Linux a way to mount network drive (as if it was connected physically to your PC https://www.dropbox.com/install-linux).

This way you can create a Local Catalog.

The benefits I see is you need not to code anything to make it work (besides the fact that you have to set up the Daemon for Dropbox to provide you with this Network drive)

I do not know if you can see any other drawbacks in this approach, but the only one I can think of is that you need to shell access to install such a thing in your machine (not everyone has a hosting where shell access is provided)

If it were necessary to implement the API v2 a feasible solution is using [Flysystem](https://flysystem.thephpleague.com/)

**Comment by wagnered** [comment]
@Leonardo,

I will look into migrating the dropbox catalog module to V2 API specs. 
No promise as to when. I'm also working on converting Ampache to the
Laravel framework for better code organization.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by wagnered** [comment]
Hopefully, someone with a better understanding of dropbox can help me understand a few things.

I created a dropbox app on dropbox then created a music catalog with the app key and app secret.

I then tried making a movies catalog with the same app key, but ampache refused to create it with the same app key.

I then created a new dropbox app, but while trying to create a movie catalog with the new app key and secret. I get "access denied".

At first, I thought that maybe a new folder must be associated with the new dropbox app, but I don't understand how to do that from my dropbox account.

Could it be because the pc hosting the ampache server is already linked to the other app?

Also, I think that ampache should be able to create different catalog types using the same app key.

I welcome any ideas about this and thanks...

**Comment by wagnered** [comment]
Found the problem: Ampache currently ignores video files.

**Comment by wagnered** [comment]
I just upgraded the Ampache Dropbox catalog to V2 but only for music at this time. It now can grab artwork, clean, and verify files.  This was done in the latest  commit #(07bf9fd).

With V2, all keys and secret can be generated and retrieved when creating the app.

I will look into adding video capability.

**Comment by wagnered** [comment]
Sorry, had to revert for a serious bug.

**Comment by wagnered** [comment]
@leonardo-chronicles  and all,

>I guess an alternative to using the Catalog Type Dropbox, instead, is to use the Catalog Type Local.

Actually, with V2, an app can be downloaded/installed on Linux and Windows to create a virtual dropbox folder mounted locally.  A local catalog of the dropbox folder might be able to be created and used like other local catalogs.

Click on "Install" from the account menu while logged in to your dropbox account.

Could anyone give this a try?

**Comment by leonardo-chronicles** [comment]
yes. I tested it and it works.
No probs with it.
The only "drawback" that I found and can be considered as such depending on your needs, is that the network drive you locally mount for dropbox syncs everything you have on your HDD if I am not wrong which means that your server's will eventually download all the data of Dropbox onto your server's HDD

**Comment by wagnered** [comment]
@Leonardo,


I see what you mean.  I just tried installing the app. The problem is
that it is not mounted as a network drive, but a subfolder created in
the home (~) folder that is synced to one's Dropbox account. I thought
it would be an actual network mounted drive that didn't need to be
synced and would be unmounted when the dropbox app is stopped.


I'll soon have the V2 dropbox-catalog at a point to be able to commit
the changes. No changes to the database are required.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by leonardo-chronicles** [comment]
@wagnered Indeed. At the time of writing I thought it was a network drive but as I confirmed later it eas not ans I forgot to clarify that.

Btw, how does your implementation of Dropbox v2 work?

1. Streams the content of dropbox to the client using Ampache as a proxy.

2. Downloads the file temporarily and after the clients finishes listening to the music is deleted.

3. The client has direct access to the dropbox path where the file is hosted.

Thanks for your contribution. Looking forward to it.

**Comment by wagnered** [comment]
@Leonardo,

For the web player, it downloads each file to a temporary location
during play and PHP deletes each one automatically when the script ends.
I have no idea how much latency there will be when playing a file from
the internet.  With localhost it is fairly fast.

Streaming to a multimedia player such as VLC may be done as time allows.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by wagnered** [comment]
@leonardo-chronicles 

I pushed the new dropbox catalog to the develop branch.  I successfully added a dropbox music and video catalog to my own server.  The latency for the music files wasn't too bad. But the videos had to completely download before starting play. There was a considerable wait for a 900mb video file to start playing.

The Laravel based version will take advantage of the html-5 audio/video player and submit the files directly to the webpage. so the player can grab the remote file directly.
